doc/src/site/sphinx/deployment/sw_google_cloud_dataproc.rst

```diff
@@ -2,180 +2,135 @@ Running Sparkling Water on Google Cloud Dataproc
+            SPARKLING_WATER_JAR='sparkling-water-assembly_2.12-3.36.0.1-1-3.1-all.jar'
 
-	cd /usr/lib/
-	wget http://h2o-release.s3.amazonaws.com/sparkling-water/spark-SUBST_SPARK_MAJOR_VERSION/SUBST_SW_VERSION/sparkling-water-SUBST_SW_VERSION.zip
-	unzip sparkling-water-SUBST_SW_VERSION.zip
+        Copy the example job source into a file named SparklingWaterGcpExampleJob.scala
 
-	pip install pip==9.0.3
-	pip install requests==2.18.4
-	pip install tabulate
-	pip install future
-	pip install scikit-learn==0.19.1
-	pip install https://h2o-release.s3.amazonaws.com/h2o/rel-SUBST_H2O_RELEASE_NAME/SUBST_H2O_BUILD_NUMBER/Python/h2o-SUBST_H2O_VERSION-py2.py3-none-any.whl
-	pip install --upgrade google-cloud-bigquery
-	pip install --upgrade google-cloud-storage
+        .. code:: scala
 
-	pip install h2o_pysparkling_SUBST_SPARK_MAJOR_VERSION
+            import java.net.URI
+            import ai.h2o.sparkling._
+            import org.apache.spark.SparkFiles
+            import org.apache.spark.sql.SparkSession
 
+            object SparklingWaterGcpExampleJob extends App {
+              // start the cluster
+              val spark = SparkSession.builder.appName("Sparkling water example").getOrCreate()
+              import spark.implicits._
+              val hc = H2OContext.getOrCreate()
 
-5. Create a Dataproc Custom Image as defined `here <https://cloud.google.com/dataproc/docs/guides/dataproc-images>`__. Note that this references the Daisy path, the custom image name, and the customization script from previous steps. The code will be similar to the following:
+              val expectedClusterSize = 3
+              val clusterSize = hc.getH2ONodes().length
+              require(clusterSize != expectedClusterSize, s"H2O cluster should be of size $expectedClusterSize but is $clusterSize")
 
- ::
+              // prepare the data
+              spark.sparkContext.addFile("https://raw.githubusercontent.com/h2oai/sparkling-water/master/examples/smalldata/prostate/prostate.csv")
+              val frame = H2OFrame(new URI("file://" + SparkFiles.get("prostate.csv")))
+              val sparkDF = hc.asSparkFrame(frame).withColumn("CAPSULE", $"CAPSULE" cast "string")
+              val Array(trainingDF, testingDF) = sparkDF.randomSplit(Array(0.8, 0.2))
 
-   python generate_custom_image.py \
-    --image-name <custom_image_name> \
-    --dataproc-version <Cloud Dataproc version (example: "1.2.22")> \
-    --customization-script <local path to your custom script> \
-    --daisy-path <local path to the downloaded daisy binary> \
-    --zone <Compute Engine zone to use for the location of the temporary VM> \
-    --gcs-bucket <URI (gs://bucket-name) of a Cloud Storage bucket in your project> \
-    [--no-smoke-test: <"true" or "false" (default: "false")>
+              // train the model
+              import ai.h2o.sparkling.ml.algos.H2OXGBoost
+              val estimator = new H2OXGBoost().setLabelCol("CAPSULE")
```

Comment:
Can we use a different algorithm? `H2ODRF`,  `H2OGBM`? `H2OXGBoost` needs additional native memory. Usually we configure [additional properties](https://docs.h2o.ai/sparkling-water/3.1/latest-stable/doc/ml/sw_xgboost.html#xgboost-memory-configuration) when running `H2OXGBoost`.